app/src/main/java/com/android/solvit/provider/ui/profile/ModifyProviderInformation.kt

```diff
@@ -0,0 +1,350 @@
+      onLocationQueryChange = { locationViewModel.setQuery(it) },
+      showDropdownLocation = showDropdown,
+      onShowDropdownLocationChange = { showDropdown = it },
+      locationSuggestions = locationSuggestions.filterNotNull(),
+      onLocationSelected = { selectedLocation = it },
+      requestLocation = null,
+      backgroundColor = colorScheme.background,
+      isValueOk = okNewLocation,
+      testTag = "newLocationInputField")
+
+  LanguageDropdownMenu(
+      selectedLanguages = newLanguage,
+      onLanguageSelected = { language, isChecked ->
+        newLanguage =
+            if (isChecked) {
+              newLanguage + language
+            } else {
+              newLanguage - language
+            }
+      })
+
+  Spacer(modifier = Modifier.height(10.dp))
+
+  Button(
+      onClick = {
+        if (allIsGood) {
+          provider.companyName = newCompanyName
+          provider.service = newProfession
+          provider.phone = newPhoneNumber
+          provider.location = selectedLocation ?: provider.location
+          provider.languages = newLanguage
+
+          listProviderViewModel.updateProvider(provider = provider)
+
+          navigationActions.goBack()
+        } else {
+          Toast.makeText(
+                  context,
+                  "Please fill in all the correct information before modify it",
+                  Toast.LENGTH_SHORT)
+              .show()
+        }
+      },
+      modifier =
+          Modifier.fillMaxWidth()
+              .height(50.dp)
+              .background(
+                  brush =
+                      if (allIsGood) {
+                        Brush.horizontalGradient(
+                            colors = listOf(colorScheme.primary, colorScheme.secondary))
+                      } else {
+                        Brush.horizontalGradient(
+                            colors =
+                                listOf(colorScheme.onSurfaceVariant, colorScheme.onSurfaceVariant))
+                      },
+                  shape =
+                      RoundedCornerShape(
+                          25.dp,
+                      )),
+      colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent)) {
+        Text(
+            "Save !", color = colorScheme.onPrimary, fontWeight = FontWeight.Bold, fontSize = 16.sp)
+      }
+
+  Spacer(modifier = Modifier.height(3.dp))
+
+  Text(
+      text = "Don't forget to save your changes by clicking the button before leaving the page!",
+      color = colorScheme.onSurfaceVariant,
+      fontSize = 12.sp,
+      textAlign = TextAlign.Center,
+      style = TextStyle(fontSize = 12.sp, lineHeight = 16.sp),
+      modifier = Modifier.padding(top = 4.dp).fillMaxWidth())
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ServiceDropdownMenu(selectedService: Services, onServiceSelected: (Services) -> Unit) {
+  var expanded by remember { mutableStateOf(false) }
+  val servicesList = Services.entries
+
```

Comment:
Isn't it easier to reuse the `ServiceTypeDropdown` from the `RequestComposables.kt` file? This would avoid unnecessary creation of new composables and code duplication.

Reply:
Yes, well done, I also tried to do this before implementing a whole new method but I was not able to use this method correctly so I created one that specifically met my needs. I'll see if in the future I can change that to make the code cleaner but I think for now we can leave it like that.